cognite/client/_api/files.py

```diff
@@ -612,19 +612,55 @@ def retrieve_download_urls(
         results = tasks_summary.joined_results(unwrap_fn=lambda res: res.json()["items"])
         return {result.get("id") or result["externalId"]: result["downloadUrl"] for result in results}
 
+    @staticmethod
+    def _create_unique_file_names(file_names_in: list[str] | list[Path]) -> list[str]:
+        """
+        Create unique file names by appending a number to the base file name.
+        """
+        file_names: list[str] = [str(file_name) for file_name in file_names_in]
+        file_names_unique: list = []
+        file_names_original_count: dict = {}
+
+        for file_name in file_names:
+            if file_name not in file_names_unique:
+                file_names_unique.append(file_name)
+                continue
+
+            file_base = file_name
+            file_ext = ""
+            if "." in file_name:
+                file_name_split = file_name.split(".")
+                file_base = ".".join(file_name_split[:-1])
+                file_ext = f".{file_name_split[-1]}"
```

Comment:
I think you are better off using `pathlib` for this with suffix (lots of good methods on `Path`) 😄

Reply:
Pathlib it is :)